atom/browser/api/atom_api_session.cc

```diff
@@ -452,14 +452,33 @@ void Session::SetCertVerifyProc(v8::Local<v8::Value> val,
 
 void Session::SetPermissionRequestHandler(v8::Local<v8::Value> val,
                                           mate::Arguments* args) {
-  AtomPermissionManager::RequestHandler handler;
-  if (!(val->IsNull() || mate::ConvertFromV8(args->isolate(), val, &handler))) {
+  using StatusCallback =
+      base::RepeatingCallback<void(blink::mojom::PermissionStatus)>;
+  using RequestHandler =
```

Comment:
I see that this redeclaration is to help use the `OnceCallback` variation of `StatusCallback` but it would be better if we can adapt the callback for repeating at the callsite of `RequestHandler` in `AtomPermissionManager` and reuse the types.

Reply:
I did it this way because I wanted to use the `OnceCallback` native_mate converter here, but I think it wasn't landed yet. (And also I wanted to try out the `AdaptRepeatingCallback` helper from Chrome)